src/rust/src/declarative_asn1/decode.rs

```diff
@@ -57,6 +59,45 @@ fn decode_printable_string<'a>(
     Ok(pyo3::Bound::new(py, PrintableString { inner })?)
 }
 
+fn decode_utc_time<'a>(
+    py: pyo3::Python<'a>,
+    parser: &mut Parser<'a>,
+) -> ParseResult<pyo3::Bound<'a, UtcTime>> {
+    let value = parser.read_element::<asn1::UtcTime>()?;
+    let dt = value.as_datetime();
+
+    let inner = crate::x509::datetime_to_py_utc(py, dt)?
+        .downcast_into::<pyo3::types::PyDateTime>()
+        .map_err(|_| {
+            pyo3::exceptions::PyValueError::new_err(
+                "error converting to datetime object".to_string(),
+            )
+        })?
```

Comment:
The downcast should never fail, so we don't need a specific error message, you can just `?`

Reply:
fixed